draft-wang-ppm-dap-taskprov.md

```diff
@@ -56,69 +60,304 @@ collector). Thus it is up to each deployment of DAP to securely implement its
+leaders and helpers can work on the same DAP task, therefore making adoption of
+DAP easier.
+
+3. A "task" now means the same task ID and same task configuration, including
+parameters for choosing VDAF. Therefore, reports from one task will only be
+aggregated with one VDAF. For the same reason, if a malicious client changes
+the task ID or task configuration, its report will be aggregated in a different
+task, with other poison reports from the same malicious attack. The "good"
+reports will not be polluted.
+
+Because extension is used in HPKE's AAD, by including all task configurations
+in the extension, malicious leader cannot change the task configuration to
+mislead helpers to reduce privacy guarantee, for e.g. by reducing
+`min_batch_size`.
+
+This extension affects upload, aggregate and collect sub-protocols.
 
 # Conventions and Definitions
 
 {::boilerplate bcp14-tagged}
 
+Similar to {{?DAP=I-D.draft-ietf-ppm-dap-01}}, this document uses the verbs
+"abort" and "alert with `[some error message]`" to describe how protocol
+participants react to various error conditions.
+
+The terms used follow those described in {{?DAP=I-D.draft-ietf-ppm-dap-01}}. The
+following new terms are used:
+
+Task provision:
+: The process of creating a DAP task.
 
-# The "task_prov" Extension
+Task configuration:
+: The non-secret parameters required to create a task in task proviosion.
+
+Task author:
+: The entity that defines the parameters of a task.
+
+
+# The "task-prov" Extension
 
 > NOTE(cjpatton) It will be useful to think of a good name for this. Something
-> succinct and descriptive is ideal. `task_prov` is fine, but we can maybe do
+> succinct and descriptive is ideal. `task-prov` is fine, but we can maybe do
 > better.
 
 A new extension is defined:
 
 ~~~
 enum {
-   task_prov(0xff00), (65535)
+    task-prov(0xff00),
+    (65535)
 } ExtensionType;
 ~~~
+> NOTE(shan) TLS uses underscore `_` to define codepoints, which is different
+> from DAP's convension of using hyphen `-`, which one to follow?
+> See [#314](ietf-wg-ppm/draft-ietf-ppm-dap#314) for discussion.
 
 When the Client includes this extension with its report, the body of the
 extension is structured as follows:
 
 ~~~
-TODO(wangshan) Specify the structure of the extension.
+struct {
+    /* Info specific for a task. */
+    opaque task_info<1..2^8-1>;
+
+    /* A list of URLs relative to which an aggregator's API endpoints */
+    /* can be found. */
+    Url aggregator_endpoints<1..2^16-1>; // Defined in I-D.draft-ietf-ppm-dap-02
+
+    /* This determines the query type for batch selection and the */
+    /* properties that all batches for this task must have. */
+    QueryConfig query_config; // Defined in I-D.draft-ietf-ppm-dap-02
+
+    /* The maximum number of times a batch of reports may be queried */
+    /* by the Collector. */
+    uint16 max_batch_lifetime;
+
+    /* Time up to which clients are allowed to upload to this task. */
```

Comment:
nit: In the VDAF spec I've adopted the convention of capitalizing Client, Aggregator, and Collector. For the DAP spec I've been pushing for doing the same. We can follow this convention or not, but we should try to be consistent.
```suggestion
    /* Time up to which Clients are allowed to upload to this task. */
```